contractcourt/chain_watcher_test.go

```diff
@@ -369,43 +366,48 @@ func TestChainWatcherDataLossProtect(t *testing.T) {
 		}
 	}
 
-	// For our first scenario, we'll ensure that if we're on state 1, and
-	// the remote party broadcasts state 2 and we don't have a pending
-	// commit for them, then we'll properly detect this as a DLP scenario.
-	if !dlpScenario(dlpTestCase{
-		BroadcastStateNum: 2,
-		NumUpdates:        1,
-	}) {
-		t.Fatalf("DLP test case failed at state 1!")
+	testCases := []dlpTestCase{
+		// For our first scenario, we'll ensure that if we're on state 1,
+		// and the remote party broadcasts state 2 and we don't have a
+		// pending commit for them, then we'll properly detect this as a
+		// DLP scenario.
+		{
+			BroadcastStateNum: 2,
+			NumUpdates:        1,
+		},
+
+		// We've completed a single update, but the remote party broadcasts
+		// a state that's 5 states byeond our best known state. We've lost
+		// data, but only partially, so we should enter a DLP secnario.
+		{
+			BroadcastStateNum: 6,
+			NumUpdates:        1,
+		},
+
+		// Similar to the case above, but we've done more than one
+		// update.
+		{
+			BroadcastStateNum: 6,
+			NumUpdates:        3,
+		},
+
+		// We've done zero updates, but our channel peer broadcasts a
+		// state beyond our knowledge.
+		{
+			BroadcastStateNum: 10,
+			NumUpdates:        0,
+		},
 	}
+	for _, testCase := range testCases {
+		testName := fmt.Sprintf("num_updates=%v,broadcast_state_num=%v",
+			testCase.NumUpdates, testCase.BroadcastStateNum)
 
-	// For the remainder of the tests, we'll perform 10 iterations with
-	// random values. We limit this number as set up of each test can take
-	// time, and also it doing up to 255 state transitions may cause the
-	// test to hang for a long time.
-	//
-	// TODO(roasbeef): speed up execution
-	err := quick.Check(dlpScenario, &quick.Config{
-		MaxCount: 10,
-		Values: func(v []reflect.Value, rand *rand.Rand) {
-			// stateNum will be the random number of state updates
-			// we execute during the scenario.
-			stateNum := uint8(rand.Int31())
-
-			// From this state number, we'll draw a random number
-			// between the state and 255, ensuring that it' at
-			// least one state beyond the target stateNum.
-			broadcastRange := rand.Int31n(int32(math.MaxUint8 - stateNum))
-			broadcastNum := uint8(stateNum + 1 + uint8(broadcastRange))
-
-			testCase := dlpTestCase{
-				BroadcastStateNum: broadcastNum,
-				NumUpdates:        stateNum,
+		t.Run(testName, func(t *testing.T) {
+			t.Parallel()
+
+			if !dlpScenario(t, testCase) {
+				t.Fatalf("test %v failed", testName)
```

Comment:
will this ever be executed? it seems the test would fail inside `dlpScenario` now

Reply:
`Errorf` fails at the very end, rather then immediately.

Reply:
gotcha, looks like there're some other calls to `Fatalf` in `dlpScenario`, though they have more to do with initialization rather than the assertions themselves. probably fine to leave it then